### PR TITLE
Introduce catalog function to print GP_TABLESPACE_VERSION_DIRECTORY

### PR DESCRIPTION
**Pull request by soumyadeep2007**
GP_TABLESPACE_VERSION_DIRECTORY's value varies with the catalog version
and is something tablespace related tests rely on. Thus we introduce
gp_tablespace_version_directory_name(). Further, this function exposes
this constant to external utilities where we anticipate its need.

Previously we had the non-catalog function equivalent called
get_tablespace_version_directory_name() which was part of the regress
framework. This commit tears it out and replaces it with the new
function.

Co-authored-by: Taylor Vesely <tvesely@pivotal.io>

## Notes
Needs a backport for 6x.

### CODE REVIEW

**Comment by asimrp** [comment]
I'm not able to see how will this function will be useful to utilities or end-users?  I would rather like to see this PR include the change to the utilities, so that the picture is clear.

File: src/backend/utils/adt/misc.c [hunk 1]
```diff
@@ -421,6 +421,15 @@ pg_tablespace_location(PG_FUNCTION_ARGS)
...
 }
 
+/*
+ * Return tablespace version directory name. GPDB_MAJORV_CATALOGV_NUMBUMPS
+ */
+Datum
+gp_tablespace_version_directory_name(PG_FUNCTION_ARGS)
+{
+	PG_RETURN_TEXT_P(CStringGetTextDatum(GP_TABLESPACE_VERSION_DIRECTORY));
+}
```
Comment (danielgustafsson) [inline src/backend/utils/adt/misc.c]: If this is required by tests, why do we need it in the catalog and not as a C UDF in the regression tests?

**Comment by jimmyyih** [comment]
Agree with Asim here.  We need more information in the commit message or PR on what/who will be using this besides tests.

**Comment by soumyadeep2007** [comment]
@tvesely had thought of a few cases where utilities external to GPDB could use it. However, we have reached consensus that we don't need it.

Thus, closing the PR.

**Pull request closed without merging**